x/farming/keeper/proposal_handler.go

```diff
@@ -58,15 +58,18 @@ func (k Keeper) AddPublicPlanProposal(ctx sdk.Context, proposals []*types.AddReq
 				p.EpochAmount,
 			)
 
+			if err = msg.ValidateBasic(); err != nil {
+				return err
+			}
+
 			plan, err := k.CreateFixedAmountPlan(ctx, msg, farmingPoolAddrAcc, terminationAcc, types.PlanTypePublic)
 			if err != nil {
 				return err
 			}
 
 			logger := k.Logger(ctx)
 			logger.Info("created public fixed amount plan", "fixed_amount_plan", plan)
-
-		} else if p.EpochRatio.IsPositive() {
+		} else if p.IsForRatioPlan() { // TODO: if all validations were correct, we can use `else` here
```

Comment:
I also think we can use `else` here as you comment it since there is validation that checks if one of `epoch amount` or `epoch ratio` is provided.